src/bots/unvetter.py

```diff
@@ -0,0 +1,130 @@
+import logging
+from typing import Callable
+
+from schema import Schema, Or
+from web3.types import BlockData
+
+import variables
+from blockchain.typings import Web3
+from cryptography.verify_signature import compute_vs
+from metrics.metrics import UNEXPECTED_EXCEPTIONS
+from metrics.transport_message_metrics import message_metrics_filter
+from transport.msg_providers.kafka import KafkaMessageProvider
+from transport.msg_providers.rabbit import RabbitProvider, MessageType
+from transport.msg_storage import MessageStorage
+from transport.msg_types.ping import PingMessageSchema, to_check_sum_address
+from transport.msg_types.unvet import UnvetMessageSchema, get_unvet_messages_sign_filter, UnvetMessage
+from transport.types import TransportType
+
+
+logger = logging.getLogger(__name__)
+
+
+class UnvetterBot:
+
+    fully_initialized = False
+
+    def __init__(self, w3: Web3):
+        self.w3 = w3
+
+        if self.w3.lido.deposit_security_module.__class__.__name__ == 'DepositSecurityModuleContractV2':
+            self.finalize_init()
+            fully_initialized = True
+
+    def finalize_init(self):
+        transports = []
+
+        if TransportType.RABBIT in variables.MESSAGE_TRANSPORTS:
+            transports.append(RabbitProvider(
+                client='unvetter',
+                routing_keys=[MessageType.UNVET, MessageType.PAUSE],
+                message_schema=Schema(Or(UnvetMessageSchema, PingMessageSchema)),
+            ))
+
+        if TransportType.KAFKA in variables.MESSAGE_TRANSPORTS:
+            transports.append(KafkaMessageProvider(
+                client=f'{variables.KAFKA_GROUP_PREFIX}unvet',
+                message_schema=Schema(Or(UnvetMessageSchema, PingMessageSchema)),
+            ))
+
+        if not transports:
+            logger.warning({'msg': 'No transports found', 'value': variables.MESSAGE_TRANSPORTS})
+
+        unvet_prefix = self.w3.lido.deposit_security_module.get_unvet_message_prefix()
+
+        self.message_storage = MessageStorage(
+            transports,
+            filters=[
+                message_metrics_filter,
+                to_check_sum_address,
+                get_unvet_messages_sign_filter(unvet_prefix),
+            ],
+        )
+
+    def execute(self, block: BlockData) -> bool:
+        if not self.fully_initialized:
+            if self.w3.lido.deposit_security_module.__class__.__name__ == 'DepositSecurityModuleContractV2':
```

Comment:
Check unveting bot proper start and blockchain state in separate functions with comments